railties/lib/rails/application/default_middleware_stack.rb

```diff
@@ -18,7 +18,7 @@ def build_stack
           middleware.use ::Rack::Sendfile, config.action_dispatch.x_sendfile_header
 
           if config.serve_static_files
-            middleware.use ::ActionDispatch::Static, paths["public"].first, config.static_cache_control
+            middleware.use ::ActionDispatch::Static, paths["public"].first, config.static_cache_control, config.action_dispatch.asset_headers
           end
```

Comment:
Will need to use `config.public_file_server.*` config here, and introduce deprecations either on `Rails::Application::Configuration#serve_static_files=` and `#static_cache_control=` (they're attr accessors now), or handle deprecations here, where the config is read. Probably better to get a deprecation warning when they're set rather than when they're read.